spec/guess_checker_spec.rb

```diff
@@ -0,0 +1,110 @@
+require_relative '../lib/guess_checker'
+
+RSpec.describe GuessChecker do
+  let(:checker) { GuessChecker.new("RGYO") }
+
+  describe "#check_red" do
+    context "when guess matches some colors in code" do
+      it "returns 2 red pins for repeated colors" do
+        expect(checker.check_red("RRRR", "RROO")).to eq(2)
+      end
+
+      xit "replaces matched colors with dashes in check code" do
+        checker.check_red("ROYG", "RGYO")
+        expect(checker.check_code).to eq("-G-O")
+      end
+
```

Comment:
Try not getting into the habit of pushing up pending tests (they tend to get forgotten ;) )